python/prog_edu_assistant_tools/prog_edu_assistant_tools/magics.py

```diff
@@ -0,0 +1,186 @@
+import io
+import re
+import sys
+import unittest
+
+from IPython.core.display import HTML
+from IPython.core.magic import (Magics, magics_class, line_magic, cell_magic,
+                                line_cell_magic)
+from jinja2 import Template
+from prog_edu_assistant_tools.summary_test_result import SummaryTestResult
+from pygments.formatters import HtmlFormatter
+from pygments import highlight
+from pygments.lexers import PythonLexer
+from types import SimpleNamespace
+
+
+def autotest(testClass):
+    """Non-magic version of the autotest.
+
+    This function can be used as
+
+        result, log = autotest(MyTestCase)
+
+    """
+    suite = unittest.TestLoader().loadTestsFromTestCase(testClass)
+    errors = io.StringIO()
+    result = unittest.TextTestRunner(verbosity=4,
+                                     stream=errors,
+                                     resultclass=SummaryTestResult).run(suite)
+    return result, errors.getvalue()
+
+
+def report(template, **kwargs):
+    """Non-magic version of the report.
+
+    This function can be used as
+
+        report(template, submission_source.source, results)
```

Comment:
Should this document that the kwargs are passed into template.render?

Reply:
Yep, good idea. Done.